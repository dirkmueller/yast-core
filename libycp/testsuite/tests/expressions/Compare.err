Parsed:
----------------------------------------------------------------------
"void"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((nil == nil) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"boolean"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((true == false) == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((true != false) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((true == true) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((false != false) == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((true && false) == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((true || false) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"integer"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((42 == 42) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((42 == -42) == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((-42 == -42) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((-42 != 42) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((((42 / 7) == (4 + 2)) == ((42 / 7) == (4 + 2))) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((((42 / 7) < (4 + 2)) == ((42 / 7) < (4 + 2))) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((((42 / 7) > (4 + 2)) == ((42 / 7) > (4 + 2))) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((((42 / 7) <= (4 + 2)) == ((42 / 7) <= (4 + 2))) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((((42 / 7) >= (4 + 2)) == ((42 / 7) >= (4 + 2))) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"float"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(((0. - 4.2) != 4.2) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((100. == (1e+10 * 1e-08)) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((42. > 42.) == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((42. >= 42.) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((42. < 42.) == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((42. <= 42.) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((42. == 42.) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((42. != 42.) == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"string"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("gr\366\337er" < "kleiner") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("gr\366\377er" < "kleiner") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("gr\366\337er" <= "kleiner") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("gr\366\337er" > "kleiner") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("klein" > "kleiner") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("klein" >= "kleiner") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("klein" == "kleiner") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("Klein" == "Klein") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("Klein" != "Klein") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(("Klein" == "klein") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((("string" + 100) == "string100") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((("string" + .path) == "string.path") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"path"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((((. + .current) + .path) == .current.path) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((((. + .current) + .path) == .current.path) != false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"list"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((["composed string", nil] == [("composed " + "string"), {
    // integer number
    // filename: "tests/expressions/Compare.ycp"
    integer number = 5;
}
]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((["composed string", nil] != [("composed " + "string"), {
    // integer number
    // filename: "tests/expressions/Compare.ycp"
    integer number = 5;
}
]) == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(([] == []) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(([1] == [1]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(([1] != [2]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(([1, 2, 3] > [1, 2]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(([1, 2] > [1, 1, 1]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(([1, "string"] > [1, 1, 1]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(([1, "string_long"] > [1, "string", 1]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(([1] < [{
    // integer number
    // filename: "tests/expressions/Compare.ycp"
    integer number = 2;
    return number;
}
]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(([1] > [{
    // integer number
    // filename: "tests/expressions/Compare.ycp"
    integer number = 2;
}
]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"term"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((`alpha () == `alpha ()) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((`alpha () < `alpha (1)) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((`alpha (1) == `alpha (1)) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((`alpha (1) < `beta (1)) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((`alpha (1) < `alpha (2)) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((`alpha (1) < `alpha (1, 1)) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((`alpha (1) < `alpha (1, "string")) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((`alpha (1) < {
    // term b
    // filename: "tests/expressions/Compare.ycp"
    term b = `beta (1);
    return b;
}
) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
((`alpha (1) > {
    // term b
    // filename: "tests/expressions/Compare.ycp"
    term b = `beta (1);
}
) == nil)
----------------------------------------------------------------------
[Interpreter] tests/expressions/Compare.ycp:103 Nil can be compared only for equality and non-equality
Parsed:
----------------------------------------------------------------------
"map"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(($[] == $[]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(($["a":1] == $["a":1]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(($["a":1] > $[]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(($["a":1, "b":2] < $["a":1, "b":3]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(($["a":1, "c":2] > $["a":1, "b":3]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(($["a":1, "b":2] > $["z":9]) == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"locale"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/expressions/Compare.ycp"
    textdomain "locale";
    return (_("a") < _("b"));
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/expressions/Compare.ycp"
    textdomain "locale";
    return (_("a") < _("aa"));
}
----------------------------------------------------------------------
[Scanner] tests/expressions/Compare.ycp:119 Warning: Empty locale constant
Parsed:
----------------------------------------------------------------------
{
    // filename: "tests/expressions/Compare.ycp"
    textdomain "locale";
    return ("" < _("a"));
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"nil"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // string val
    // filename: "tests/expressions/Compare.ycp"
    string val = nil;
    return (val == nil);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // string val
    // filename: "tests/expressions/Compare.ycp"
    string val = nil;
    return ! (val != nil);
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // string val
    // filename: "tests/expressions/Compare.ycp"
    string val = nil;
    return ! (val == "foo");
}
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
{
    // string val
    // filename: "tests/expressions/Compare.ycp"
    string val = nil;
    return (val != "foo");
}
----------------------------------------------------------------------
