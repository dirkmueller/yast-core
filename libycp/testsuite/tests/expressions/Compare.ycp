
# ---------------------------------------------------------
#
#  Filename:    Compare.ycp
#
#  Purpose:     test cases for YCP test
#
#  Creator:     jd@suse.de
#
#  Maintainer:  jd@suse.de
#
# ---------------------------------------------------------

# comparisons
"void"
# void
(( nil == nil ) == true )

# boolean
"boolean"
((true == false) == false )
((true != false) == true )
((true == true) == true )
((false != false) == false )
(( true && false ) == false )
(( true || false ) == true )

# integer
"integer"
(( 42 == 42 ) == true )
(( 42 == -42) == false )
((-42 == -42) == true )
((-42 != 42 ) == true )
(((42 / 7 == 4 + 2 ) == (( 42 / 7 ) == (4 + 2))) == true )
(((42 / 7 <  4 + 2 ) == (( 42 / 7 ) <  (4 + 2))) == true )
(((42 / 7 >  4 + 2 ) == (( 42 / 7 ) >  (4 + 2))) == true )
(((42 / 7 <= 4 + 2 ) == (( 42 / 7 ) <= (4 + 2))) == true )
(((42 / 7 >= 4 + 2 ) == (( 42 / 7 ) >= (4 + 2))) == true )

# float
"float"
(((0.0-4.2) != 42.0e-1 ) == true )
((100.0 == 1.0e10 * 10.0e-9) == true )
(( 42.0 >  4.2e1 ) == false )
(( 42.0 >= 4.2e1 ) == true )
(( 42.0 <  4.2e1 ) == false )
(( 42.0 <= 4.2e1 ) == true )
(( 42.0 == 4.2e1 ) == true )
(( 42.0 != 4.2e1 ) == false )

# strings
"string"
(( "größer" < "kleiner" ) == true )
(( "gr\366\377er" < "kleiner" ) == true )
(( "größer" <= "kleiner" ) == true )
(( "größer" > "kleiner" ) == false )
(( "klein"  > "kleiner" ) == false )
(( "klein"  >= "kleiner" ) == false )
(( "klein"  == "kleiner" ) == false )
(( "Klein"  == "Klein" ) == true )
(( "Klein"  != "Klein" ) == false )
(( "Klein"  == "klein" ) == false )
((( "string" + 100 ) == "string100" ) == true )
((( "string" + .path ) == "string.path" ) == true )

# paths
"path"
(((. + .current + .path) == .current.path ) == true )
(((. + .current + .path) == .current.path ) != false )

# locale
# change within locale: the string is now a constant
# not tested any longer --> moved to unsolved

# list
"list"
(( [ "composed string", nil ] == [ "composed " + "string", { integer number = 5; } ] ) == true )
(( [ "composed string", nil ] != [ "composed " + "string", { integer number = 5; } ] ) == false )

#list comparison   added by tom@suse.de
(( [ ] == [ ] ) == true )
(( [ 1 ] == [ 1 ] ) == true )
(( [ 1 ] != [ 2 ] ) == true )
(( [ 1, 2, 3 ] > [ 1, 2 ] ) == true )
(( [ 1, 2 ] > [ 1, 1, 1 ] ) == true )
(( [ 1, "string" ] > [ 1, 1, 1 ] ) == true )
(( [ 1, "string_long" ] > [ 1, "string", 1 ] ) == true )
(( [ 1 ] < [ { integer number = 2; return number; } ] ) == true )
(( [ 1 ] > [ { integer number = 2; } ] ) == true )

# term
# there is already a test within D_DefineTerm.ycp
"term"
# term comparison    added by tom@suse.de
(( `alpha() == `alpha() ) == true )
(( `alpha() < `alpha( 1 ) ) == true )
(( `alpha( 1 ) == `alpha( 1 ) ) == true )
(( `alpha( 1 ) < `beta( 1 ) ) == true )
(( `alpha( 1 ) < `alpha( 2 ) ) == true )
(( `alpha( 1 ) < `alpha( 1, 1 ) ) == true )
(( `alpha( 1 ) < `alpha( 1, "string" ) ) == true )
(( `alpha( 1 ) < { term b = `beta( 1 ); return b; } ) == true )
(( `alpha( 1 ) > { term b = `beta( 1 ); } ) == true )

# map comparison     added by tom@suse.de
"map"
(( $[] == $[] ) == true )
(( $[ "a":1 ] == $[ "a":1 ] ) == true )
(( $[ "a":1 ] > $[] ) == true )
(( $[ "a":1, "b":2 ] < $[ "a":1, "b":3 ] ) == true )
(( $[ "a":1, "c":2 ] > $[ "a":1, "b":3 ] ) == true )
(( $[ "a":1, "b":2 ] > $[ "z":9 ] ) == true )


#Locale comparison     added by tom@suse.de
"locale"
{ textdomain "locale"; return ( _("a") < _("b") ); }
{ textdomain "locale"; return ( _("a") < _("aa") ); }
{ textdomain "locale"; return ( _("") < _("a") ); }
