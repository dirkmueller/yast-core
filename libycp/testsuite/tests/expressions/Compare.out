("void")
(true)
("boolean")
(true)
(true)
(true)
(true)
(true)
(true)
("integer")
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
("float")
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
("string")
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
("path")
(true)
(true)
("list")
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
("term")
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
("map")
(true)
(true)
(true)
(true)
(true)
(true)
("locale")
(true)
(true)
(true)
("nil")
(true)
(true)
(true)
(true)
